hive: Add hive.Command()

joamaki opened the pull request:
hive.Command() now implements the hive related cobra commands, so they don't need to be repeated per binary. See commits for more details.

```
$ go run ./daemon hive
Cells:

  Ⓜ️ agent (Cilium Agent):
      Ⓜ️ infra (Infrastructure):
          Ⓜ️ gops (Gops Agent):
              ⚙️ (gops.GopsConfig) {
                  GopsPort: (uint16) 9890
              }
...

$ go run ./operator hive
Cells:
...

$ go run ./daemon hive dot-graph | dot -Tx11

```

joamaki committed:
hive: Fix InfoLeaf line wrapping

The line wrapping was still broken when there were multiple wrapped lines,
since after wrapping the current line length was miscalculated.

Signed-off-by: Jussi Maki <jussi@isovalent.com>

joamaki committed:
hive: Add hive cobra command

Implement the commands for hive object and dot-graph dumping
centrally, so we'll avoid duplicate implementations in each
binary. Add the hive as sub-command to agent, operator and
clustermesh-apiserver.

Add a test for the agent cell to check that it can be instantiated
with the default configuration so we don't break the ability to
inspect the agent hive.

Remove endpoint-gc-interval flag from option.Config as it already
was in endpointmanager.Config. Looks like use of PersistentFlags()
allowed duplicates. PersistentFlags no longer needed as the hive
sub-command now takes the hive's flag set (so it doesn't need to inherit
from root command).

Signed-off-by: Jussi Maki <jussi@isovalent.com>

joamaki committed:
operator: Fix operator hive population with default config

LBIPAM nil-deref'd when k8s configuration was not provided, fix
by adding a nil check. While at it, move lbipam cell definitions
to cell.go.

Add test to operator to check that the operator hive can be populated.

Signed-off-by: Jussi Maki <jussi@isovalent.com>

pippolo84 reviewed (approved):
LGTM :100:

tklauser commented on daemon/cmd/daemon_main.go:
```diff
@@ -882,10 +882,6 @@ func initializeFlags() {
 	flags.Int(option.EndpointQueueSize, defaults.EndpointQueueSize, "Size of EventQueue per-endpoint")
 	option.BindEnv(Vp, option.EndpointQueueSize)
 
-	flags.Duration(option.EndpointGCInterval, 5*time.Minute, "Periodically monitor local endpoint health via link status on this interval and garbage collect them if they become unhealthy, set to 0 to disable")
-	flags.MarkHidden(option.EndpointGCInterval)
-	option.BindEnv(Vp, option.EndpointGCInterval)
```
If possible, maybe split out the flag removal into a dedicated commit? I saw the note in the commit message about it, but it looked a bit odd/unrelated in the diff.

joamaki committed:
option/config: Remove leftover EndpointGCInterval

Remove endpoint-gc-interval flag from option.Config as it already
was in endpointmanager.Config. Looks like use of PersistentFlags()
allowed duplicates. PersistentFlags no longer needed as the hive
sub-command now takes the hive's flag set (so it doesn't need to inherit
from root command).

Signed-off-by: Jussi Maki <jussi@isovalent.com>
